dimod/higherorder/utils.py

```diff
@@ -336,11 +336,12 @@ def make_quadratic(poly: Mapping[Tuple[Variable, Variable], Bias], strength: flo
     return bqm
 
 
-def poly_energy(sample_like, poly: Mapping[Tuple[Variable, Variable], Bias]) -> float:
+def poly_energy(sample_like: SamplesLike,
+                poly: Mapping[Tuple[Variable, Variable], Bias]) -> float:
```

Comment:
This should be `Mapping[Tuple[Variable, ...]], Bias]` since the tuples are variable length. Also, you can just do
```python
Poly = Mapping[Tuple[Variable, ...]], Bias]
```
somewhere at the top if you like and then use it everywhere.

Reply:
Also, technically I think it's a sequence, so 
```python
Poly = Mapping[Sequence[Variable]], Bias]
```

Reply:
Could also consider adding this new type to `dimod.typing` and using it from there.